aws/rust-runtime/aws-auth/src/provider/cache.rs

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::provider::CredentialsResult;
+use crate::Credentials;
+use std::future::Future;
+use std::sync::Arc;
+use std::time::{Duration, SystemTime};
+use tokio::sync::{OnceCell, RwLock};
+
+#[derive(Clone)]
+pub(super) struct Cache {
+    margin: Duration,
```

Comment:
can you add a doc comment explaining what this field is?

Reply:
I think building time based expiry into this cache explicitly might not be exactly what we want, but fine to keep it this way for now

Reply:
Added comment and renamed to `buffer_time`.

Hard to say right now, but I think alternate caching credentials provider implementations will use the cache the same way. Seems like a safe bet to defer refactoring until then.